### PR TITLE
Migrate all jsx2 to jsx3

### PR DESCRIPTION
**Pull request by maxlibin**
This is a begging of migration of sketch client from jsx2 to jsx3, the idea is migrate parts by part and still keep client working as is with `wrapReactForReasonReact` where it requires jsx2 support.

- [ ] Migrate all stateless components to jsx3 (wrapReactForReasonReact)
- [ ] Migrate all reducer components to jsx3 (wrapReactForReasonReact)
- [ ] Remove all wrapReactForReasonReact

### CODE REVIEW

File: client/src/NotFound.re [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+      </section>
+    </main>
+  </div>;
+};
+
+module Jsx2 = {
+  let component = ReasonReact.statelessComponent(__MODULE__);
+
+  let make = children => {
+    let children = React.array(children);
```
Comment (andrii-antiukhov) [inline client/src/NotFound.re:26]: @maxlibin I see originally there are no children in component
`let make = (~entity="path", _children) => {`
Why you add children for JSX3 version? @jchavarri Sorry, am i correct here? I need to know too :-) Thanks!

File: client/src/components/Logo.re [hunk 2]
```diff
@@ -14,6 +15,20 @@ let make = (~size=1., ~className=?, _children) => {
...
-    </svg>;
-  },
+    </svg>
+  </div>;
+};
+
+module Jsx2 = {
+  let component = ReasonReact.statelessComponent(__MODULE__);
+
+  let make = (~size: float, ~className: option(string)=?, children) => {
```
Comment (andrii-antiukhov) [inline client/src/components/Logo.re:25]: @jchavarri @maxlibin Do we need children here?

File: client/src/components/UI_FullpageLoading.re [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
+    <Logo size=0.3 className="UI_FullpageLoading--logo" />
+    <UI_DotFlashing />
+  </div>;
+};
+
+module Jsx2 = {
+  let component = ReasonReact.statelessComponent(__MODULE__);
+
+  let make = children => {
+    let children = React.array(children);
```
Comment (andrii-antiukhov) [inline client/src/components/UI_FullpageLoading.re:17]: Same, do we need here children? @maxlibin

File: client/src/NotFound.re (hunk 1 above)
Comment (maxlibin) [reply client/src/NotFound.re:26]: you mean adding to jsx 2, I might be wrong but I assume we always need pass in `_children` in Jsx 2 even is not used.

Comment (jchavarri) [reply client/src/NotFound.re:26]: As discussed, let's try to migrate all components at once, so there is no need to write any `Jsx2`, which is super confusing. I started a PR with bsconfig configured to use jsx3 in #321, and fixed already a few components.

@max @andrii-antiukhov To make the code easier to review, you can open PRs against the branch in #321, e.g. https://github.com/Sketch-sh/sketch-sh/compare/jsx3...maxlibin:feature/jsx3?expand=1

Thanks!

Comment (jchavarri) [reply client/src/NotFound.re:26]: > To make the code easier to review

There is no need to finish a folder or anything before opening a PR. I largely prefer multiple small PRs often, than one huge PR, so reviewing is easier, and also we avoid 2 people doing the same work twice :)

**Comment by maxlibin** [comment]
since we decided to migrate at once: https://github.com/Sketch-sh/sketch-sh/pull/321, will close this PR.

**Pull request closed without merging**